### PR TITLE
#80 Private post in inbox, deleted print

### PR DESCRIPTION
**Pull request by HolaClubSandwich**

### CODE REVIEW

File: social_distribution/posts/views.py [hunk 1]
```diff
@@ -57,7 +57,17 @@ def post_create(request, author_id):
                 friends = author.followers.all() & author.followings.all()
                 for friend in friends:
                     friend.inbox.posts.add(post)
-            # TODO: inbox private posts
+            elif post.visibility == "private":
+                friends = author.followers.all() & author.followings.all()
+                # if users are friend
+                for friend in friends:
+                    if post.visibleTo == str(friend.user):
```
Comment (vhnguyen0707) [inline social_distribution/posts/views.py]: - If you want to create private post for friends and followers only. You should be checking if the recipient is valid when you attempt to create the post already, not when you’re done creating post then inbox.
- You shouldn’t be converting an user object into string like this. A more Django and pythonic way is using user.username

File: social_distribution/posts/views.py [hunk 2]
```diff
@@ -57,7 +57,17 @@ def post_create(request, author_id):
                 friends = author.followers.all() & author.followings.all()
                 for friend in friends:
                     friend.inbox.posts.add(post)
-            # TODO: inbox private posts
+            elif post.visibility == "private":
+                friends = author.followers.all() & author.followings.all()
+                # if users are friend
```
Comment (vhnguyen0707) [inline social_distribution/posts/views.py]: - Friends are also followers so  if you do 2 for loops like this, there will inherently be duplicates.
- It is not very optimal to traverse through the whole queryset, and check if the usernames are matching, just get author object of the recipient and check if it is in the followers list.

**Review by vhnguyen0707 — CHANGES_REQUESTED** [review]

File: social_distribution/posts/views.py (hunk 1 above)
Comment (HolaClubSandwich) [reply social_distribution/posts/views.py]: User can still create private post to non friend or unfollowed users tho?

Comment (vhnguyen0707) [reply social_distribution/posts/views.py]: Then why would you inbox to friends or followers only? If the recipient is not friend or follower, the only way to see the post is having the uri or it shows up on new feed. But there are lots of public and friend posts from local and the other 3 remote nodes. It is more even unlikely to see it and it’s not a good design UX-wise.

Comment (HolaClubSandwich) [reply social_distribution/posts/views.py]: I thought inbox is only showing friends and followers post update?

Comment (HolaClubSandwich) [reply social_distribution/posts/views.py]: Oh right, my bad, I understand it now

**Review by vhnguyen0707 — APPROVED** [review]

**Pull request merged**